### PR TITLE
Adds User Token Details page

### LINKED ISSUE #7789: User Tokens Details
#### ISSUE TYPE
 - Feature Idea

##### SUMMARY
This issue is about adding user token details.
A proposed flow is here: https://tower-mockups.testing.ansible.com/patternfly/users/users-tokens-detail/

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
Fixes #7789 
This is a new flow which adds a details page to Tokens.  
There is 1 known issue with this work and it is tracked by this issue https://github.com/ansible/awx/issues/7797

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION
Screen Shot 2020-08-03 at 1 15 45 PM

### CODE REVIEW

**Comment by trahman73** [comment]
@AlexSCorey Can we capitalize the Scope value?

File: awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <CardBody>
+      <DetailList>
+        <Detail
+          label={i18n._(t`Application`)}
+          value={summary_fields?.application?.name}
+          dataCy="team-detail-name"
+        />
+        <Detail label={i18n._(t`Description`)} value={description} />
+        <Detail label={i18n._(t`Scope`)} value={scope} />
+        <Detail label={i18n._(t`Created`)} value={formatDateString(created)} />
```
Comment (nixocio) [inline awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx]: Hey @AlexSCorey, is there any particular reason to use `Detail` over `UserDateDetail`?

image

File: awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import ErrorDetail from '../../../components/ErrorDetail';
+import { formatDateString } from '../../../util/dates';
+import { TokensAPI } from '../../../api';
+import useRequest, { useDismissableError } from '../../../util/useRequest';
+
+function UserTokenDetail({ token, canEditOrDelete, i18n }) {
+  const { scope, description, created, modified, summary_fields } = token;
+  const history = useHistory();
+  const { id, tokenId } = useParams();
+  const { request: deleteTeam, isLoading, error: deleteError } = useRequest(
```
Comment (nixocio) [inline awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx]:
```suggestion
  const { request: deleteToken, isLoading, error: deleteError } = useRequest(
```

File: awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  const { error, dismissError } = useDismissableError(deleteError);
+
+  return (
+    <CardBody>
+      <DetailList>
+        <Detail
+          label={i18n._(t`Application`)}
+          value={summary_fields?.application?.name}
+          dataCy="team-detail-name"
```
Comment (nixocio) [inline awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx]:
```suggestion
          dataCy="application-token-detail-name"
```

**Review by nixocio — CHANGES_REQUESTED** [review]
I got this code locally. Changes seem to working as described. Suggested minor changes. Good job @AlexSCorey.

File: awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx (hunk 1 above)
Comment (AlexSCorey) [reply awx/ui_next/src/screens/User/UserTokenDetail/UserTokenDetail.jsx]: no, I should have used that.  Thanks for catching that.

**Comment by jakemcdermott** [comment]
@nixocio @AlexSCorey Is this one ready for merge or QE?

**Comment by nixocio** [comment]
I would say QE. @dsesami already self requested a review. I will add the`needs_test` flag.

**Review by nixocio — APPROVED** [review]

**Comment by dsesami** [comment]
Identified some issues with @AlexSCorey where the most recent changes make the applications modal behave strangely. alex is pushing a fix.

**Comment by dsesami** [comment]
This looks much better. hyperlinks work, the token creation flow works with or without an associated application.

**Review by dsesami — APPROVED** [review]

**Pull request merged**